### ISSUE TITLE
A better textual representation for diffing layout tables

### ISSUE THREAD

**Issue by cmyr** [googlefonts/fontc#544]
So we're finding it very hard to read the ttx-based diffs that we are currently using to compare the output of `fontc` and `fontmake`, and we would like to figure out an alternative.

Ideally this alternative would address the following:
- go fetch deltas from the `ItemVariationStore` and display them inline, instead of just showing the `VariationIndex` table values
- display just the rules (we don't care how they're split into subtables or lookups)
  - ... except how then do we indicate the lookup flags or the filtering sets?
  - ... and we're going to just be okay with losing some potentially important info in not retaining subtables
- display the rules with some sort of canonical sorting order
- group rules by language system, duplicating any rules in lookups referenced by multiple language systems
- come up with some simple, readable representation of the various rule types
- ... figure out how to represent contextual rules??

**Comment by rsheeter** [comment]
> we don't care how they're split into subtables or lookups

https://learn.microsoft.com/en-us/typography/opentype/spec/gpos#table-organization says to `apply the lookups in the order given in the LookupList table.` which makes me think order would matter if two rules can collide with one winner (rather than both applying, but sometimes the effects don't seem to combine well (two kerning adjustments? sure. two anchor positions? less sure))

**Comment by khaledhosny** [comment]
Lookup order definitely matters. Also how rules are organized into lookups matters as well, the same two rules in one lookup can work differently than when split into two lookups.

**Comment by cmyr** [comment]
yes, we are essentially deciding that for the purposes of our diff tool we are not going to account for this (important in practice!) detail, since if we do then we lose the whole benefit of this idea, I think?

**Comment by khaledhosny** [comment]
But this would mask possible regressions in lookup order and splitting (especially that what goes into what lookup is unnecessarily complicated in FEA syntax).

**Comment by rsheeter** [comment]
Naively optimistic hat on, I'm hoping we can account for order mapping while eliminating noise from cases where it doesn't. For example, if two binaries record how A and combining breve fit together in ways that are binary different but semantically equivalent can we convert that to a normalized form that would compare equal?

**Comment by rsheeter** [comment]
Potentially useful ref, a kerning normalizer from Behdad, https://github.com/behdad/playground/blob/master/eigenkernfea/eigenkernfea.py

**Comment by cmyr** [comment]
added in #578
